core/src/main/java/com/graphhopper/routing/util/MountainBikeFlagEncoder.java

```diff
@@ -128,6 +128,16 @@ protected MountainBikeFlagEncoder(int speedBits, double speedFactor, int maxTurn
         preferHighwayTags.add("residential");
         preferHighwayTags.add("unclassified");
 
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.EXCELLENT, 1.1d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.GOOD, 1.0d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.INTERMEDIATE, 0.9d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.BAD, 0.7d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.VERY_BAD, 0.6d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.HORRIBLE, 0.5d);
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.VERY_HORRIBLE, 0.4d);
+        // SmoothnessSpeed <= smoothnessFactorPushingSectionThreshold gets mapped to speed PUSHING_SECTION_SPEED
+        setSmoothnessSpeedFactor(com.graphhopper.routing.ev.Smoothness.IMPASSABLE, smoothnessFactorPushingSectionThreshold);
```

Comment:
One last question: do we need to overwrite this here if already set in BikeFlagEncoder?

Reply:
As MountainBikeFlagEncoder extends BikeCommonFlagEncoder and not BikeFlagEncoder I would say yes!

Reply:
Ah, ok :+1: